### PR TITLE
Install XS9 with XenRT

### PR DESCRIPTION
**Pull request by liulinC**

### CODE REVIEW

**Comment by liulinC** [comment]
Job install succeed: 3901109

File: repository.py [hunk 1]
```diff
@@ -704,7 +704,7 @@ def access(self, path):
             # couldn't parse the server name out:
             return False
 
-    def openAddress(self, address):
+    def openAddress(self, address, mode="rb"):
```
Comment (freddy77) [inline repository.py:707]: It looks like from description and usage that the mode should be always binary.
Here you are adding an ignored parameter.

File: util.py [hunk 2]
```diff
@@ -264,11 +264,9 @@ def fetchFile(source, dest):
...
                 request = url.getPlainURL()
 
-            fd = urllib.request.urlopen(request)
-            fd_dest = open(dest, 'w')
-            shutil.copyfileobj(fd, fd_dest)
-            fd_dest.close()
-            fd.close()
+            with urllib.request.urlopen(request) as fd_resp:
+                with open(dest, 'wb') as fd_dest:
+                    shutil.copyfileobj(fd_resp, fd_dest)
```
Comment (freddy77) [inline util.py:269]: Why not using a comma like
```python
    with urllib.request.urlopen(request) as fd_resp, open(dest, 'wb') as fd_dest:
        shutil.copyfileobj(fd_resp, fd_dest)
```

File: repository.py [hunk 3]
```diff
@@ -694,7 +704,7 @@ def access(self, path):
...
```
[the remark this answers is not in the document]
Comment (liulinC) [reply repository.py:707]: yeah, we can drop one of the place.

**Review by GeraldEV — APPROVED** [review]

**Comment by liulinC** [comment]
job: 3910127 passed.

**Review by freddy77 — DISMISSED** [review]

**Comment by liulinC** [comment]
- XenRT: 3916169
- Manual test from main.iso

**Review by freddy77 — APPROVED** [review]
Fine with it

**Review by GeraldEV — APPROVED** [review]

**Pull request merged**